### DISCUSSION TITLE
Firebase Login fails with valid id token

### DISCUSSION THREAD

**Discussion by inf0rmatix** [serverpod/serverpod#910] [Q&A]
I wanted to use Firebase Auth.
After hacking through the rather incomplete documentation,
I found out, that for some reason, client.modules.auth.firebase.authenticate(idToken.token!) fails.

I have checked the token, it is a valid JWT, The payload has the email in it. The only log I get is "Created Auth object", after that, the execution ends and the client receives
{"success":false,"key":null,"keyId":null,"userInfo":null,"failReason":0}

Serverpod version 1.1.0

**Comment by inf0rmatix** [comment]
After digging into the code, I printed the exception from firebase admin sdk which states:
```
FirebaseAppError(app/invalid-credential): Credential implementation provided to initializeApp() via the "credential" property failed to fetch a valid Google OAuth2 access token with the following error: "OpenIdException(invalid_grant): Invalid JWT Signature.". There are two likely causes: (1) your server time is not properly synced or (2) your certificate key file has been revoked. To solve (1), re-sync the time on your server. To solve (2), make sure the key ID for your key file is still present at https://console.firebase.google.com/iam-admin/serviceaccounts/project. If not, generate a key file at https://console.firebase.google.com/project/_/settings/serviceaccounts/adminsdk.
```

**Comment by inf0rmatix** [comment]
Ok, seems the problem solved itself. 

I don't know why, but what I did solved it:

1. Deleted the service account key in Google IAM (those where generated through firebase)
2. Created a new key directly in Google IAM for the firebase admin sdk account
3. ???
4. Login works, however fetching/creating the user doesn't. Users.findByEmail throws "null check operator used on a null value"

**Comment by vlidholt** [comment]
We are looking into this now!

**Reply by inf0rmatix** [reply]
Hi, sorry for missing out to update you guys.

The second issue I mentioned was something on my side. Had to delete the docker volumes and recreate everything.

However, it would be very good to not have Firebase Authentication fail silently but rather get the error message in the server response :)
